mesh/mesh.hpp

```diff
@@ -777,10 +777,9 @@ class Mesh
    long GetGlobalNE() const { return ReduceInt(NumOfElements); }
 
    /** @brief Return the mesh geometric factors corresponding to the given
-       integration rule. If @a use_temp is true and using device memory the
-       GeometricFactors will be constructed in temporary memory. */
+       integration rule. */
    const GeometricFactors* GetGeometricFactors(const IntegrationRule& ir,
-                                               const int flags, bool use_temp=false);
+                                               const int flags);
```

Comment:
We'll need an option to pass in a `MemoryType`, did you have something else in mind?

Reply:
I was thinking of temporarily switching the device memory type: e.g. we can have a macro (or some c++ function with lambdas magic) like this:
```c++
MFEM_WITH_DEVICE_MEMORY(MemoryType::DEVICE_UMPIRE_2,
   auto geom_factors = mesh->GetGeometricFactors(ir, flags);
);
```
We can also add a `MemoryType` option for the device memory but then for the default value we'll need to add another constant in the `MemoryType` enum -- something like `MemoryType::DEVICE_DEFAULT` which will tell the method to use the currently set device memory type.

Reply:
This could work! Neat idea!

Reply:
I do like `MFEM_WITH_DEVICE_MEMORY`